### PR TITLE
[DOCS] Remove `PrintCompressedOopsMode` JVM option

### LINKED ISSUE #71644: [DOCS] Guidance for zero based compressed oops and removal of PrintCompressedOopsMode
The documentation ([link](https://www.elastic.co/guide/en/elasticsearch/reference/master//advanced-configuration.html)) suggests to use `PrintCompressedOopsMode` but this setting was removed in JDK 15 at a certain point (https://bugs.openjdk.java.net/browse/JDK-8245723), e.g. the one bundled with Elasticsearch 7.9.3 docker images.
```
openjdk version "15" 2020-09-15
OpenJDK Runtime Environment (build 15+36-1562)
OpenJDK 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)
```

We get:
```
Unrecognized VM option 'PrintCompressedOopsMode'
```

It seems those informations are now logged as INFO with `-Xlog:gc,gc+init`.

In `jvm.options`, we use the following:
```
9-:-Xlog:gc*,gc+age=trace,safepoint:file=${loggc}:utctime,pid,tags:filecount=32,filesize=64m
```

Which shows already the information we need:
```
[2021-04-13T16:21:41.187+0000][5778][gc,init]  Compressed Oops: Enabled (...
```

We might need `-Xlog:gc+heap+coops` to show the information about alignment.

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:
* Removes a reference to the
  `-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` JVM option. This
  option is no longer supported.
* Combines `Xms/Xmx` recommendations for compressed oops.

Closes #71644.

### Preview
https://elasticsearch_73741.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/advanced-configuration.html#set-jvm-heap-size

### CODE REVIEW

File: docs/reference/setup/advanced-configuration.asciidoc [hunk 1]
```diff
@@ -110,35 +110,16 @@ NOTE: When running in a container, such as <<docker,Docker>>, total memory is
 defined as the amount of memory visible to the container, not the total system
 memory on the host.
 
-* Set `Xms` and `Xmx` to no more than 32 GB, the approximate threshold for
-compressed ordinary object pointers (oops). To verify you are under the
+* Set `Xms` and `Xmx` to no more than the threshold for compressed ordinary
+object pointers (oops). The exact threshold varies but 26GB is safe on most
+systems and can be as large as 30GB on some systems. To verify you are under the
 threshold, check `elasticsearch.logs` for an entry like this:
```
Comment (DJRickyB) [inline docs/reference/setup/advanced-configuration.asciidoc:116]: while we're here, `elasticsearch.logs` isn't anything right?
```suggestion
threshold, check the {es} log for an entry like this:
```

Comment (jrodewig) [reply docs/reference/setup/advanced-configuration.asciidoc]: Thanks for catching this. This should have been `elasticsearch.log` originally, which is in `$ES_HOME/logs` for our tar archive distros. I like {es} logs better though.

**Review by danielmitterdorfer — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
Thanks @DJRickyB @danielmitterdorfer.

**Pull request merged**